client/src/BibleEngineClient.ts

```diff
@@ -22,6 +23,43 @@ export class BibleEngineClient {
         if (bibleEngineOptions) this.localBibleEngine = new BibleEngine(bibleEngineOptions);
     }
 
+    static chunk(arr: any, chunkSize = 1, cache: any[] = []) {
+        const tmp = [...arr];
+        if (chunkSize <= 0) return cache;
+        while (tmp.length) cache.push(tmp.splice(0, chunkSize));
+        return cache;
+    }
+
+    static async getBookIndexFile(versionUid: string, baseS3Url: string) {
+        const fileIndexUrl = `${baseS3Url}/${versionUid}/index.json`;
+        const response = await fetch(fileIndexUrl);
+        const fileIndex = await response.json();
+        return fileIndex;
+    }
+
+    async downloadVersion(versionUid: string, baseS3Url: string) {
+        console.time('downloadVersion');
+        const books = await BibleEngineClient.getBookIndexFile(versionUid, baseS3Url);
```

Comment:
i'd suggest adding the option to download the data from the api if no `baseUrl` is given. i think we would need to add the possibility to request full books from the API, which is trivial.

Reply:
removed all the download code, I'm going to do a followup PR with tests and the like